### ISSUE TITLE
Issue using AppCenter 5.0 preview on iOS - .NET MAUI

### ISSUE THREAD

**Issue by 94mc** [microsoft/appcenter-sdk-dotnet#1692]
### **Description**

When i am adding the AppCenter code to MauiProgram.cs , it is breaking. I have added a try catch to the code and will post both the code and the exception below.

**1. Code sample**
AppCenter.Start("ios=mysecret" +
                  "windowsdesktop={Your UWP App secret here};" +
                  "android=mysecret;" +
                  "macos={Your macOS App secret here};",
                  typeof(Analytics), typeof(Crashes));

**2. Exception**
{System.Exception: Could not create an native instance of the type 'Microsoft.AppCenter.Apple.Bindings.MSACWrapperSdk': the native class hasn't been loaded.
It is possible to ignore this condition by setting ObjCRuntime.Class.ThrowOnInitFailure to false.
   at Foundation.NSObject.InitializeObject(Boolean alloced)
   at Foundation.NSObject..ctor(NSObjectFlag x)
   at Microsoft.AppCenter.Apple.Bindings.MSACWrapperSdk..ctor(String wrapperSdkVersion, String wrapperSdkName, String wrapperRuntimeVersion, String liveUpdateReleaseLabel, String liveUpdateDeploymentKey, String liveUpdatePackageHash)
   at Microsoft.AppCenter.AppCenter.SetWrapperSdk()
   at Microsoft.AppCenter.AppCenter.PlatformStart(String appSecret, Type[] services)
   at Microsoft.AppCenter.AppCenter.Start(String appSecret, Type[] services)
   at AppName.MauiProgram.CreateMauiApp() in C:\Users\myfilepath...

I have seen a workaround posted below which some people have had luck with but unfortunately the issue persists. It is changing this setting to true:
![image](https://user-images.githubusercontent.com/40669752/196694371-d70da2fe-9b00-4d91-a4d0-74516d49d059.png)

### **Details**

1. What is your app platform (Xamarin.Android or Xamarin.iOS or UWP)?
    .NET MAUI
2. If using Xamarin.Forms or if using portable/shared code to call our SDK APIs, are you using shared project, PCL code or .NET standard code for the application? Which .NET standard version or which PCL profile?
     .NET 6.0
2. Which SDK version are you using?
    5 Preview
4. Which OS version did you experience the issue on?
     iOS 16
5. What device version did you see this error on?  Were you using an emulator or a physical device?
    iPhone 13 Pro Max physical device

**Comment by DmitriyKirakosyan** [comment]
Hi @94mc , thank you for reporting this issue! Can you share the dotnet version you build the app with? Run `dotnet --version` in your project root dir.
Are you building the app on Windows or macOS? What other targets are present in your csproj file?

Would it be possible for you to reproduce the issue on an empty project and share it with us?

**Comment by 94mc** [comment]
Hi there, after running that command , dotnet version: **7.0.100-rc.2.22477.23**
I am building on **Windows** to a physical iOS device running iOS 16
Targets in csproj (if im understanding correctly, apologies if not) are :
![image](https://user-images.githubusercontent.com/40669752/197187254-722a6401-4a06-4542-9708-a551075efc4e.png)
![image](https://user-images.githubusercontent.com/40669752/197187297-f450fe7f-a65c-494e-b0b8-ec75cfd01790.png)

I will try and make time to reproduce on an empty project also but i thought id send on this info in the mean time

**Comment by EdwardMcFlurry** [comment]
> ### **Description**
> When i am adding the AppCenter code to MauiProgram.cs , it is breaking. I have added a try catch to the code and will post both the code and the exception below.
> 
> **1. Code sample** AppCenter.Start("ios=mysecret" + "windowsdesktop={Your UWP App secret here};" + "android=mysecret;" + "macos={Your macOS App secret here};", typeof(Analytics), typeof(Crashes));
> 
> **2. Exception** {System.Exception: Could not create an native instance of the type 'Microsoft.AppCenter.Apple.Bindings.MSACWrapperSdk': the native class hasn't been loaded. It is possible to ignore this condition by setting ObjCRuntime.Class.ThrowOnInitFailure to false. at Foundation.NSObject.InitializeObject(Boolean alloced) at Foundation.NSObject..ctor(NSObjectFlag x) at Microsoft.AppCenter.Apple.Bindings.MSACWrapperSdk..ctor(String wrapperSdkVersion, String wrapperSdkName, String wrapperRuntimeVersion, String liveUpdateReleaseLabel, String liveUpdateDeploymentKey, String liveUpdatePackageHash) at Microsoft.AppCenter.AppCenter.SetWrapperSdk() at Microsoft.AppCenter.AppCenter.PlatformStart(String appSecret, Type[] services) at Microsoft.AppCenter.AppCenter.Start(String appSecret, Type[] services) at AppName.MauiProgram.CreateMauiApp() in C:\Users\myfilepath...
> 
> I have seen a workaround posted below which some people have had luck with but unfortunately the issue persists. It is changing this setting to true: ![image](https://user-images.githubusercontent.com/40669752/196694371-d70da2fe-9b00-4d91-a4d0-74516d49d059.png)
> 
> ### **Details**
> 1. What is your app platform (Xamarin.Android or Xamarin.iOS or UWP)?
>    .NET MAUI
> 2. If using Xamarin.Forms or if using portable/shared code to call our SDK APIs, are you using shared project, PCL code or .NET standard code for the application? Which .NET standard version or which PCL profile?
>    .NET 6.0
> 3. Which SDK version are you using?
>    5 Preview
> 4. Which OS version did you experience the issue on?
>    iOS 16
> 5. What device version did you see this error on?  Were you using an emulator or a physical device?
>    iPhone 13 Pro Max physical device

Do a clean solution, preferably delete bin & obj folders and clean the solution again as that will work.

**Comment by 94mc** [comment]
> > ### **Description**
> > When i am adding the AppCenter code to MauiProgram.cs , it is breaking. I have added a try catch to the code and will post both the code and the exception below.
> > **1. Code sample** AppCenter.Start("ios=mysecret" + "windowsdesktop={Your UWP App secret here};" + "android=mysecret;" + "macos={Your macOS App secret here};", typeof(Analytics), typeof(Crashes));
> > **2. Exception** {System.Exception: Could not create an native instance of the type 'Microsoft.AppCenter.Apple.Bindings.MSACWrapperSdk': the native class hasn't been loaded. It is possible to ignore this condition by setting ObjCRuntime.Class.ThrowOnInitFailure to false. at Foundation.NSObject.InitializeObject(Boolean alloced) at Foundation.NSObject..ctor(NSObjectFlag x) at Microsoft.AppCenter.Apple.Bindings.MSACWrapperSdk..ctor(String wrapperSdkVersion, String wrapperSdkName, String wrapperRuntimeVersion, String liveUpdateReleaseLabel, String liveUpdateDeploymentKey, String liveUpdatePackageHash) at Microsoft.AppCenter.AppCenter.SetWrapperSdk() at Microsoft.AppCenter.AppCenter.PlatformStart(String appSecret, Type[] services) at Microsoft.AppCenter.AppCenter.Start(String appSecret, Type[] services) at AppName.MauiProgram.CreateMauiApp() in C:\Users\myfilepath...
> > I have seen a workaround posted below which some people have had luck with but unfortunately the issue persists. It is changing this setting to true: ![image](https://user-images.githubusercontent.com/40669752/196694371-d70da2fe-9b00-4d91-a4d0-74516d49d059.png)
> > ### **Details**
> > 
> > 1. What is your app platform (Xamarin.Android or Xamarin.iOS or UWP)?
> >    .NET MAUI
> > 2. If using Xamarin.Forms or if using portable/shared code to call our SDK APIs, are you using shared project, PCL code or .NET standard code for the application? Which .NET standard version or which PCL profile?
> >    .NET 6.0
> > 3. Which SDK version are you using?
> >    5 Preview
> > 4. Which OS version did you experience the issue on?
> >    iOS 16
> > 5. What device version did you see this error on?  Were you using an emulator or a physical device?
> >    iPhone 13 Pro Max physical device
> 
> Do a clean solution, preferably delete bin & obj folders and clean the solution again as that will work.

I tried this and no luck unfortunately. The error persists

**Comment by DmitriyKirakosyan** [comment]
@94mc  This looks like a net6-ios issue on windows. Do you have a mac machine to try to build your project on it?

**Comment by plattski** [comment]
I appear to be having the same problem. I am unable to get App Center to work with my iOS app in MAUI. Just creating the basic MAUI app, adding the Nuget packages, adding the call to AppCenter.Start, crashes it every time. 

I'm using .NET 6, building in Windows, on Microsoft Visual Studio Community 2022 (64-bit) - Preview, Version 17.4.0 Preview 5.0, connecting directly to the iPad

Using iPad with iOS v15.6.1 Sample code online at https://drive.google.com/file/d/15pIPuWaYemGzMFZHIlBupamsrI0lOyNC/view?usp=share_link. Deugger console output reads:

[0:] An error occurred: 'Could not create an native instance of the type 'Microsoft.AppCenter.Apple.Bindings.MSACWrapperSdk': the native class hasn't been loaded.
It is possible to ignore this condition by setting ObjCRuntime.Class.ThrowOnInitFailure to false.'. Callstack: ' at Foundation.NSObject.InitializeObject(Boolean alloced)
at Foundation.NSObject..ctor(NSObjectFlag x)
at Microsoft.AppCenter.Apple.Bindings.MSACWrapperSdk..ctor(String wrapperSdkVersion, String wrapperSdkName, String wrapperRuntimeVersion, String liveUpdateReleaseLabel, String liveUpdateDeploymentKey, String liveUpdatePackageHash)
at Microsoft.AppCenter.AppCenter.SetWrapperSdk()
at Microsoft.AppCenter.AppCenter.PlatformStart(String appSecret, Type[] services)
at Microsoft.AppCenter.AppCenter.Start(String appSecret, Type[] services)
at AppCenterTest.App.OnStart() in D:\AppCenterTest\AppCenterTest\App.xaml.cs:line 21
at Microsoft.Maui.Controls.Application.SendStart()
at Microsoft.Maui.Controls.Window.Microsoft.Maui.IWindow.Created()
at Microsoft.Maui.LifecycleEvents.AppHostBuilderExtensions.<>c.b__1_0(UIApplication app, NSDictionary launchOptions)
at Microsoft.Maui.MauiUIApplicationDelegate.<>c__DisplayClass6_0.b__0(FinishedLaunching del)
at Microsoft.Maui.LifecycleEvents.LifecycleEventServiceExtensions.InvokeLifecycleEvents[FinishedLaunching](IServiceProvider services, Action`1 action)
at Microsoft.Maui.MauiUIApplicationDelegate.FinishedLaunching(UIApplication application, NSDictionary launchOptions)
at UIKit.UIApplication.UIApplicationMain(Int32 argc, String[] argv, IntPtr principalClassName, IntPtr delegateClassName)
at UIKit.UIApplication.Main(String[] args, Type principalClass, Type delegateClass)
at AppCenterTest.Program.Main(String[] args) in D:\AppCenterTest\AppCenterTest\Platforms\iOS\Program.cs:line 13
at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)'
The app has been terminated.

**Comment by hareshkalyani** [comment]
I am getting same error.

I am not creating MAUI app, Its Xamarin Native project with .Net6 Environment. 

AppCenter SDK is 5.0.1.

Any fixes to this?

**Comment by aleksandr-dorofeev** [comment]
Hey @hareshkalyani 
Сould you provide a reproduction of the issue on a sample project?

**Comment by swetaVanjara** [comment]
any fixes ??? I'm facing same problem.

**Comment by AlleSchonWeg** [comment]
If anybody in this thread use a physical iPhone with VS on windows, than the cause of this issue is this: https://github.com/microsoft/appcenter-sdk-dotnet/issues/1680
VS HotRestart can only work with dynamic libs. But the sdk uses static libs. Unfortunately my feature request to change this has no votes.

**Comment by frogcrush** [comment]
I'm having this issue as well on vs4mac. 

![image](https://user-images.githubusercontent.com/6116333/237019569-43f69b46-5341-4378-bc3b-4dd4b095c094.png)

![image](https://user-images.githubusercontent.com/6116333/237019963-af4be802-c58e-458c-9b68-197b5037c7ee.png)

iOS Simulator version is 16.4 and Xcode version is 14.3.

**Comment by MikhailSuendukov** [comment]
Hi @swetaVanjara, could you please share more details, and point on what system you are working with Visual Studio.

**Comment by MikhailSuendukov** [comment]
I am closing this issue due to inactivity. If you have any questions, you can freely reopen it.

**Comment by ozersenol** [comment]
I have same error MAUI Blazor App with Real device Debug Mod  iOS 16.6 and VS 2022 Version 17.7.3 Mac is connected and all provisioning is set as it should be. I tried deleting obj and bin folders, Reseting even test device. 

[0:] An error occurred: 'Could not create an native instance of the type 'Microsoft.AppCenter.Apple.Bindings.MSACWrapperSdk': the native class hasn't been loaded.
It is possible to ignore this condition by setting ObjCRuntime.Class.ThrowOnInitFailure to false.'. Callstack: '   at Foundation.NSObject.InitializeObject(Boolean alloced)
   at Foundation.NSObject..ctor(NSObjectFlag x)
   at Microsoft.AppCenter.Apple.Bindings.MSACWrapperSdk..ctor(String wrapperSdkVersion, String wrapperSdkName, String wrapperRuntimeVersion, String liveUpdateReleaseLabel, String liveUpdateDeploymentKey, String liveUpdatePackageHash)
   at Microsoft.AppCenter.AppCenter.SetWrapperSdk()
   at Microsoft.AppCenter.AppCenter.PlatformStart(String appSecret, Type[] services)
   at Microsoft.AppCenter.AppCenter.Start(String appSecret, Type[] services)
   at Hemmengel.Mobile.MauiProgram.CreateMauiApp() in D:\Projects\Git\HemmenGel\Hemmengel.Mobile\MauiProgram.cs:line 34
   at Hemmengel.Mobile.AppDelegate.CreateMauiApp() in D:\Projects\Git\HemmenGel\Hemmengel.Mobile\Platforms\iOS\AppDelegate.cs:line 8
   at Microsoft.Maui.MauiUIApplicationDelegate.WillFinishLaunching(UIApplication application, NSDictionary launchOptions)
   at UIKit.UIApplication.UIApplicationMain(Int32 argc, String[] argv, IntPtr principalClassName, IntPtr delegateClassName)
   at UIKit.UIApplication.Main(String[] args, Type principalClass, Type delegateClass)
   at Hemmengel.Mobile.Program.Main(String[] args) in D:\Projects\Git\HemmenGel\Hemmengel.Mobile\Platforms\iOS\Program.cs:line 13
   at System.Reflection.MethodInvoker.InterpretedInvoke(Object obj, Span`1 args, BindingFlags invokeAttr)'
The app has been terminated.

**Comment by appuviradiya** [comment]
Hey Any Update on this?

**Comment by SteveSitekitcare** [comment]
I would love an update too

**Comment by DmitriyKirakosyan** [comment]
@ozersenol , unfortunately MAUI Blazor App is not supported.

@appuviradiya , @SteveSitekitcare , please provide more details about the issue you're encountering? Are you developing the app on a Windows or Mac? Which version of .NET are you using? Furthermore, please provide a detailed description of the error messages you are encountering in the output.

**Comment by SteveSitekitcare** [comment]
> @ozersenol , unfortunately MAUI Blazor App is not supported.
> 
> @appuviradiya , @SteveSitekitcare , please provide more details about the issue you're encountering? Are you developing the app on a Windows or Mac? Which version of .NET are you using? Furthermore, please provide a detailed description of the error messages you are encountering in the output.

Hi @DmitriyKirakosyan 
Here is a sample app that has the issue (insert secret in app.xaml.cs)
https://github.com/SteveSitekitcare/AppCenter.IOS.Sample

The Error is the same as the original post creators.

`{System.Exception: Could not create an native instance of the type 'Microsoft.AppCenter.Apple.Bindings.MSACWrapperSdk': the native class hasn't been loaded.`



- Im using Windows 11 
- Im using MAUI on DotNet7
- Im debugging on a physical IOS device (iphone 13) but the app crashes on the market too. 
- If I debug using a virtual device it works.
- IOS version 16.6.1



Microsoft Visual Studio Enterprise 2022
Version 17.6.6
VisualStudio.17.Release/17.6.6+33927.249
Microsoft .NET Framework
Version 4.8.09032

Installed Version: Enterprise

Visual C++ 2022   00476-80000-00000-AA886
Microsoft Visual C++ 2022

ADL Tools Service Provider   1.0
This package contains services used by Data Lake tools

ASA Service Provider   1.0

ASP.NET and Web Tools   17.6.326.62524
ASP.NET and Web Tools

Azure App Service Tools v3.0.0   17.6.326.62524
Azure App Service Tools v3.0.0

Azure Data Lake Tools for Visual Studio   2.6.5000.0
Microsoft Azure Data Lake Tools for Visual Studio

Azure Functions and Web Jobs Tools   17.6.326.62524
Azure Functions and Web Jobs Tools

Azure Stream Analytics Tools for Visual Studio   2.6.5000.0
Microsoft Azure Stream Analytics Tools for Visual Studio

C# Tools   4.6.0-3.23329.3+5db000294db8cc634e7d8fc004844338c9747dbc
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Common Azure Tools   1.10
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

Extensibility Message Bus   1.4.21 (main@8f226a8)
Provides common messaging-based MEF services for loosely coupled Visual Studio extension components communication and integration.

Microsoft Azure Hive Query Language Service   2.6.5000.0
Language service for Hive query

Microsoft Azure Stream Analytics Language Service   2.6.5000.0
Language service for Azure Stream Analytics

Microsoft Azure Tools for Visual Studio   2.9
Support for Azure Cloud Services projects

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

Mono Debugging for Visual Studio   17.6.41 (790a401)
Support for debugging Mono processes with Visual Studio.

NuGet Package Manager   6.6.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit https://docs.nuget.org/

Razor (ASP.NET Core)   17.6.0.2327201+a6a61fdfa748eaa65aab53dab583276e26af4a3e
Provides languages services for ASP.NET Core Razor.

SQL Server Data Tools   17.6.15.1
Microsoft SQL Server Data Tools

ToolWindowHostedEditor   1.0
Hosting json editor into a tool window

TypeScript Tools   17.0.20329.2001
TypeScript Tools for Microsoft Visual Studio

Visual Basic Tools   4.6.0-3.23329.3+5db000294db8cc634e7d8fc004844338c9747dbc
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual C++ for Cross Platform Mobile Development (Android)   17.0.33606.364
Visual C++ for Cross Platform Mobile Development (Android)

Visual F# Tools   17.6.0-beta.23174.5+0207bea1afae48d9351ac26fb51afc8260de0a97
Microsoft Visual F# Tools

Visual Studio IntelliCode   2.2
AI-assisted development for Visual Studio.

VisuallChatGPTStudio   1.0
Empty VSIX Project.

VisualStudio.DeviceLog   1.0
Information about my package

VisualStudio.Mac   1.0
Mac Extension for Visual Studio

VSPackage Extension   1.0
VSPackage Visual Studio Extension Detailed Info

Xamarin   17.6.0.251 (d17-6@318364c)
Visual Studio extension to enable development for Xamarin.iOS and Xamarin.Android.

Xamarin Designer   17.6.6.0 (remotes/origin/d17-6@cb430751d1)
Visual Studio extension to enable Xamarin Designer tools in Visual Studio.

Xamarin.Android SDK   13.2.0.6 (d17-5/a200af1)
Xamarin.Android Reference Assemblies and MSBuild support.
    Mono: 6dd9def
    Java.Interop: xamarin/java.interop/d17-5@149d70fe
    SQLite: xamarin/sqlite/3.40.1@68c69d8
    Xamarin.Android Tools: xamarin/xamarin-android-tools/d17-5@9f02d77

**Comment by DmitriyKirakosyan** [comment]
@SteveSitekitcare , Thank you for the details and the demo app!
Would it be possible for you to test your app on Mac machine and check if it works?
Unfortunately, this error happens often when building the app on Windows.

You mentioned that the app crashes on market too. Does it crash consistently or time to time?

**Comment by smurfster** [comment]
> @SteveSitekitcare , Thank you for the details and the demo app! Would it be possible for you to test your app on Mac machine and check if it works? Unfortunately, this error happens often when building the app on Windows.
> 
> You mentioned that the app crashes on market too. Does it crash consistently or time to time?

Sorry I forgot to mention I use a Mac to build the app. iOS Maui apps cannot be built on windows

**Comment by matt-goldman** [comment]
I have the same issue with apps built using Visual Studio for Mac. Works fine in debug config, but in release build crashes on startup.

**Comment by SteveSitekitcare** [comment]
> I have the same issue with apps built using Visual Studio for Mac. Works fine in debug config, but in release build crashes on startup.

I;ve been in touch with support at App Center. The official line from App Center is they are concentrating on dotnet 6. Which seems a little crazy as theyre a microsoft company, you would think they would be promoting/supporting the latest version of dotnet

**Comment by matt-goldman** [comment]
That does seem strange. In the meantime I've discovered that a clean and rebuild sometimes helps. I don't fully understand it.

**Comment by Mercally** [comment]
Same here, I'm working on Mac with Visual Studio and Visual Studio Code.

I recently installed the following nugget packages to my Net Maui 8 project (platforms iOS and Android).
-Microsoft.AppCenter.Analytics v5.0.3
-Microsoft.AppCenter.Crashes v5.0.3

I called the AppCenter.Start(...) method into the App.xamml.cs class, for Android it seems to is working fine, but iOS crashes and terminate the app directly:

![Screenshot 2024-01-15 at 10 34 35 PM](https://github.com/microsoft/appcenter-sdk-dotnet/assets/29028686/899c7d0e-5b82-4706-a845-04a907a30a7f)


I would like to use the App Center to monitor and release my app.
